flink-yarn/src/test/java/org/apache/flink/yarn/YarnResourceManagerTest.java

```diff
@@ -521,4 +533,56 @@ public void testOnContainerCompleted() throws Exception {
 			});
 		}};
 	}
+
+	/**
+	 * 	Tests that YarnResourceManager will trigger to reject all pending slot request, when maximum number of failed
+	 * 	contains is hit.
+	 */
+	@Test
+	public void testOnContainersAllocatedWithFailure() throws Exception {
```

Comment:
With the generalized test for the failure rate behaviour in `ResourceManagerTest` we would only need to check that a failing container would call `ResourceManager#recordFailure`.